app/main/page.tsx

```diff
@@ -0,0 +1,158 @@
+"use client";
+import { useEffect, useState } from "react";
+import Editor from "../components/editor/editor";
+import Viewer from "../components/viewer/viewer";
+import { H1 } from "../styles/uni-classes";
+import { useDict } from "../utils/useDictHook";
+import { IResponse, getSchema, sendRequest } from "../utils/request";
+import { signOut, useSession } from "next-auth/react";
+import { ServerChooser } from "../components/server-chooser";
+import { formatCode } from "../utils/formateCode";
+import { GiComb } from "react-icons/gi";
+import { BsPlayCircle } from "react-icons/bs";
+import { useServerRequestContext } from "../context/contexts";
+import { HelpSection } from "../components/help-section";
+import { MdErrorOutline } from "react-icons/md";
+import Loader from "../components/loader";
+
+const Page = () => {
+  const { status, data: sessionData } = useSession();
+  const tokenExpiry = sessionData?.user.token_expiry;
+  useEffect(() => {
+    const isTokenExpired = tokenExpiry && new Date(tokenExpiry) < new Date();
+    if (status === "authenticated" && isTokenExpired) {
+      signOut({ callbackUrl: "/", redirect: true });
+    }
+  }, [sessionData, status, tokenExpiry]);
+  const dict = useDict();
+  const [response, setResponse] = useState<IResponse>({});
+  const { state, dispatch } = useServerRequestContext();
+  const [error, setError] = useState("");
+  const [schemaErrors, setSchemaErrors] = useState("");
+  useEffect(() => {
+    async function getData() {
```

Comment:
I believe we don't need async function here as we don't use await. So this wrap function is redundant.